rclcpp/wait_set/thread_safe_wait_set.cpp

```diff
@@ -0,0 +1,112 @@
+// Copyright 2021 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <cassert>
+#include <memory>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+int main(int argc, char * argv[])
+{
+  rclcpp::init(argc, argv);
+
+  auto node = std::make_shared<rclcpp::Node>("wait_set_example_node");
+
+  auto do_nothing = [](std_msgs::msg::String::UniquePtr) {assert(false);};
+  auto sub1 = node->create_subscription<std_msgs::msg::String>("~/chatter", 10, do_nothing);
+  auto sub2 = node->create_subscription<std_msgs::msg::String>("~/chatter", 10, do_nothing);
+  std::vector<decltype(sub1)> sub_vector = {sub1, sub2};
+  auto guard_condition1 = std::make_shared<rclcpp::GuardCondition>();
+  auto guard_condition2 = std::make_shared<rclcpp::GuardCondition>();
+
+  // FIXME: removing sub if it was added in the ctor leads to a failure
```

Comment:
@wjwwood I was getting an exception here when adding the subscription in the ctor. I guess this will be gone once https://github.com/ros2/rclcpp/pull/1700 is merged.

Reply:
Right this is one of the bugs being fixed.

For now you could add it after construction and it would work. Which is what you've done 👍